From c71c3937afebe405baa988ba3bfb816024d6144d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Thu, 05 Aug 2021 17:36:55 +0300
Subject: [PATCH] [SW-0] gaudi: security: refine qm protection

Each QMAN has a GW for indirect access.
These GW must be protected.
Protect the GW using the protection bits.

Change-Id: Ib28fe80130f20dbb19b08007fdf5fad7e8a7cd1d
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index 1f7c8f6..09d1fbc 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -536,10 +536,16 @@
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/*
-	 * offset: 0xC80 - 0xE80 - allowed access
-	 * offset: 0xE80 - 0xF80 as following:
+	 * offset: 0xC80 - 0xD00 as following:
 	 */
-	prot_base += SNGL_REG_PROT_SIZE * 5;
+	prot_base += SNGL_REG_PROT_SIZE;
+	mask = (PROT_BIT(mmTPC0_QM_IND_GW_APB_CFG) &
+		PROT_BIT(mmTPC0_QM_IND_GW_APB_WDATA) &
+		PROT_BIT(mmTPC0_QM_IND_GW_APB_RDATA) &
+		PROT_BIT(mmTPC0_QM_IND_GW_APB_STATUS));
+	regmap_writel(mask, PRIV_PROT_REG(prot_base));
+
+	prot_base += SNGL_REG_PROT_SIZE * 4;
 	set_prot_bits_range_all(prot_base, PROT_SIZE_MAX - (prot_base - base));
 }
 
